### ISSUE TITLE
`null/` shows in shields for origin after disabling all origins

### ISSUE THREAD

**Issue by bsclifton** [brave/brave-browser#1901]
Discovered when reviewing https://github.com/brave/brave-extension/pull/78

## Steps to reproduce
1. Visit vox.com
2. Use shields to block script. Page refreshes
3. Open shields, scroll down to `Toggle switches to disable script blocking`
4. Toggle all the entries and hit `Apply`
5. A mysterious `null/` entry shows up

<img width="275" alt="screen shot 2018-10-29 at 9 50 45 pm" src="https://user-images.githubusercontent.com/4733304/47696594-b9147a00-dbc4-11e8-95ae-fa7dbb5110bc.png">

## Version information
Version 0.57.3 Chromium: 70.0.3538.67 (Official Build) dev (64-bit)
(macOS 10.13.6)

**Comment by diracdeltas** [comment]
as mentioned in https://github.com/brave/brave-extension/pull/78#issuecomment-434158345, the null origin is not allowable no matter how many times one seems to click the 'allow' button.

**Comment by simonhong** [comment]
When ads & tracker is blocked, there is no `/null` entry after applying once.

**Comment by diracdeltas** [comment]
@simonhong : why would `null` always be blocked by ads and trackers? IIRC @yrliou mentioned it could be caused by `data:application/javascript;...` js sources, which are not necessarily blocked by adblocking/tp to my knowledge.

I also occasionally encounter pages that are broken with script blocking enabled which show `null` in the blocked scripts list and don't work if i allow all-scripts-except-null, only working if i allow all scripts by turning off script blocking. This seems potentially related

**Comment by simonhong** [comment]
@diracdeltas  I think `/null` is the result of blocking ads and trackers.
Let me explain why I'm thinking like that.

After loading `vox.com` with all scripts are blocked (and ads and trackers blocked is on), I enabled `https://cdn.vox-cdn.com/` in the blocked list.

Below is JS network capture.
As you can see, `chorus-74016cc9ceb291c43eef.js` is enabled but `concert_ads-38fe20caec9bb2f7937e.js` is failed to load and `/null` is added in the blocked scripts list.
<img width="922" alt="screen shot 2019-01-03 at 9 46 53 am" src="https://user-images.githubusercontent.com/6786187/50619467-b7589100-0f3c-11e9-9ec3-c5b3b4ef2359.png">

Instead it is added to the blocked list of ads and trackers like below.
<img width="379" alt="screen shot 2019-01-03 at 9 50 21 am" src="https://user-images.githubusercontent.com/6786187/50619556-259d5380-0f3d-11e9-962d-87f2ca06bddd.png">

On the other hand, two scrips are enabled when ads and trackers are disabled.
Of course, there is no `/null` entry in the blocked scripts list.

So, I guess `/null` is the result by blocking of ads and trackers.

**Comment by diracdeltas** [comment]
@simonhong interesting; here is why i think `null/` might not be ads

1. go to https://www.vox.com/the-goods/2019/1/3/18167072/apple-iphone-sales-down-china-upgrades-q1-2018 and notice at the bottom of the page that some images are not fully loaded when scripts are blocked
2. open shields, open scripts menu, allow all checked scripts
3. notice that it says 'null' is still blocked and images still don't show up if you try to allow 'null'. 
<img width="1113" alt="screen shot 2019-01-03 at 1 46 52 pm" src="https://user-images.githubusercontent.com/549654/50663299-111d8300-0f5e-11e9-9144-f9195d21413b.png">

4. now go to shields and change 'scripts blocked' to 'all scripts allowed'. notice that the images show up:
<img width="1087" alt="screen shot 2019-01-03 at 1 47 41 pm" src="https://user-images.githubusercontent.com/549654/50663344-390ce680-0f5e-11e9-8115-8f90d9167967.png">

The expected behavior is that if images show up in step 4, they should also show up in step 3 (because the result of allowing all checked scripts should be the same as turning off script blocking).

**Comment by bsclifton** [comment]
Closing in favor of https://github.com/brave/brave-browser/issues/5346

Starting with 0.68.x, it no longer shows as null. Instead, it shows and cannot be un-blocked
